src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceValueSerializer.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Reflection;
+using System.Xml.Linq;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData.Formatter.Serialization
+{
+    /// <summary>
+    /// Represents an <see cref="ODataSerializer"/> for serializing <see cref="IEdmComplexType" />'s.
+    /// </summary>
+    public class ODataResourceValueSerializer : ODataEdmTypeSerializer
```

Comment:
>ODataResourceValueSerializer [](start = 17, length = 28)

1) I think we should have a "ODataAnnotationSerializer", not a "ODataResourceValueSerializer",   @michael pizzo

2) So, we have the serializer, how about the deserializer?

Reply:
ResourceSerializer doesnt output resource value but a resource thats why created ResourceValueSerialiser. For deserializer it should be ResourceDeserializer, right?

Reply:
I agree with the current naming; this serializes ODataResourceValues -- there is technically nothing in this class specific to annotations.  It's *usage* is by annotations, because the only way you can serialize a complex or entity type in an annotation is as an ODataResourceValue, but in the future there could conceivably be other places we serialize ODataResources.


---
In reply to: [489048322](https://github.com/OData/WebApi/pull/2219#discussion_r489048322) [](ancestors = 489048322)

Reply:
See my other comments.  This serializer doesn't know anything about annotations; it just knows how to serialize complex types and entity types as resourcevalues.

---
In reply to: [489007239](https://github.com/OData/WebApi/pull/2219#discussion_r489007239) [](ancestors = 489007239)